context-core/src/main/java/de/atb/context/persistence/common/RepositoryTDB.java

```diff
@@ -31,160 +31,155 @@
-                TDB.sync(set);
-                if (set.getDefaultModel() != null) {
-                    TDB.sync(set.getDefaultModel());
-                    set.getDefaultModel().removeAll();
-                    TDB.sync(set.getDefaultModel());
-                    set.getDefaultModel().close();
-                    set.asDatasetGraph().close();
+            });
+            set.close();
+        }
+        datasets.clear();
+    }
+
+    /**
+     * (non-Javadoc)
+     *
+     * @see de.atb.context.persistence.common.Repository#reset(de.atb.context.common.util.BusinessCase)
+     */
+    @Override
+    public final boolean reset(final BusinessCase bc) {
+        final Dataset set = this.datasets.remove(bc);
+        if (set != null) {
+            transactional(set, (ds) -> {
+                TDB.sync(ds);
+                if (ds.getDefaultModel() != null) {
+                    TDB.sync(ds.getDefaultModel());
+                    ds.getDefaultModel().removeAll();
+                    TDB.sync(ds.getDefaultModel());
+                    ds.getDefaultModel().close();
+                    ds.asDatasetGraph().close();
                 }
-                set.commit();
-            } catch (Exception e) {
-                logger.error("Error occurred, rolling back.");
-                set.abort();
-            } finally {
-                set.end();
-            }
-			set.close();
-			TDB.closedown();
-		}
+            });
+            set.close();
+            TDB.closedown();
+        }
         try {
             initializeDataset(bc);
             return true;
         } catch (ConfigurationException e) {
-            logger.error(e.getMessage());
+            logger.error(e.getMessage(), e);
         }
         return false;
-	}
+    }
 
-	@Override
-	public final synchronized Dataset getDataSet(final BusinessCase bc) {
-		Dataset ds = datasets.get(bc);
-		if (ds == null) {
+    @Override
+    public final synchronized Dataset getDataSet(final BusinessCase bc) {
+        Dataset ds = datasets.get(bc);
+        if (ds == null) {
             try {
                 ds = initializeDataset(bc);
-            } catch (ConfigurationException e) {}
-		}
-		return ds;
-	}
+            } catch (ConfigurationException e) {
+                logger.error(e.getMessage(), e);
+            }
+        }
+        return ds;
+    }
+
+    protected void transactional(final Dataset dataSet, final Consumer<Dataset> block) {
+        dataSet.begin(ReadWrite.WRITE);
+        try {
+            block.accept(dataSet);
+            dataSet.commit();
+        } catch (Exception e) {
+            logger.error("Error occurred, rolling back.", e);
+            dataSet.abort();
+        } finally {
+            dataSet.end();
+        }
+    }
+
+    protected <R> R transactional(final Dataset dataSet, final R defaultResult, final Supplier<R> block) {
+        // FIXME: Should this be `begin(ReadWrite.READ)`? Because everything in `block` seems to be just "selects"?
```

Comment:
Please fix

Reply:
If this is set to RedWrite.READ, the test cases are failing, so it was set on purpose to ReadWrite.WRITE